packages/astro/src/core/errors/overlay.ts

```diff
@@ -0,0 +1,531 @@
+  font-size: 22px;
+  margin: 0;
+  padding: 24px;
+  border-bottom: 1px solid var(--border);
+}
+
+#stack-content {
+  font-size: 14px;
+  white-space: pre;
+  line-height: 21px;
+  overflow: auto;
+  padding: 24px;
+  color: var(--stack-text);
+}
+`;
+
+const overlayTemplate = /* html */ `
+<style>
+${style.trim()}
+</style>
+<div id="backdrop">
+  <div id="layout">
+    <header id="header">
+      <section id="header-left">
+        <h2 id="name"></h2>
+        <h1 id="title">An error occurred.</h1>
+      </section>
+      <div id="houston-overlay"></div>
+      <div id="houston">
+<svg xmlns="http://www.w3.org/2000/svg" width="175" height="131" fill="none"><path fill="currentColor" d="M55.977 81.512c0 8.038-6.516 14.555-14.555 14.555S26.866 89.55 26.866 81.512c0-8.04 6.517-14.556 14.556-14.556 8.039 0 14.555 6.517 14.555 14.556Zm24.745-5.822c0-.804.651-1.456 1.455-1.456h11.645c.804 0 1.455.652 1.455 1.455v11.645c0 .804-.651 1.455-1.455 1.455H82.177a1.456 1.456 0 0 1-1.455-1.455V75.689Zm68.411 5.822c0 8.038-6.517 14.555-14.556 14.555-8.039 0-14.556-6.517-14.556-14.555 0-8.04 6.517-14.556 14.556-14.556 8.039 0 14.556 6.517 14.556 14.556Z"/><rect width="168.667" height="125" x="3.667" y="3" stroke="currentColor" stroke-width="4" rx="20.289"/></svg>
+      </div>
+    </header>
+
+    <section id="message-hints">
+      <section id="message">
+        <span id="message-icon">
+          <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M12 7v6m0 4.01.01-.011M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10Z"/></svg>
+        </span>
+        <div id="message-content"></div>
+      </section>
+      <section id="hint">
+        <span id="hint-icon">
+          <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none"><path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="m21 2-1 1M3 2l1 1m17 13-1-1M3 16l1-1m5 3h6m-5 3h4M12 3C8 3 5.952 4.95 6 8c.023 1.487.5 2.5 1.5 3.5S9 13 9 15h6c0-2 .5-2.5 1.5-3.5h0c1-1 1.477-2.013 1.5-3.5.048-3.05-2-5-6-5Z"/></svg>
+        </span>
+        <div id="hint-content"></div>
+      </section>
+    </section>
+
+		<section id="code">
+			<header>
+				<h2></h2>
+			</header>
+			<div id="code-content"></div>
+		</section>
+
+    <section id="stack">
+      <h2>Stack Trace</h2>
+      <div id="stack-content"></div>
+    </section>
+  </div>
+</div>
+`;
+
+const openNewWindowIcon =
+	/* html */
+	'<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="none"><path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M14 2h-4m4 0L8 8m6-6v4"/><path stroke="currentColor" stroke-linecap="round" stroke-width="1.5" d="M14 8.667V12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h3.333"/></svg>';
+
+const { HTMLElement = class {} } = globalThis;
```

Comment:
what's this about?

Reply:
Ah, I figured that in dev we could have a neater experience by writing the overlay code directly in JS inside of inside a string and then using `toString()` on the class when injecting it

Reply:
I'm probably not understanding, but can you not just ts-ignore?

Reply:
Tested this again and I misremembered, it's not a build error but a runtime one, it'll complain that HTMLElement doesn't exist (since well, it doesn't exists on Node)

[Here's where this is originally in Vite's code](https://github.com/vitejs/vite/blob/db8df14d8934f11fa64e8b33e376f816b3b16f15/packages/vite/src/client/overlay.ts#L133-L135)

Reply:
No no, don't want to go to that trouble over this. This is a minor thing we can talk about later.

Reply:
I get it now. You are serializing the class using toString(). There are other ways to do this to avoid this code running in the browser (outside of making it a string), but not important.